File: programmers/LV2/요격 시스템.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool comp(vector<int> a, vector<int> b) {
    if(a[1] == b[1]) {
        return a[0] < b[0];
    } else {
        return a[1] < b[1];
    }
}

int solution(vector<vector<int>> targets) {
    sort(targets.begin(), targets.end(), comp);
    int ref = targets[0][1];
    int answer = 1;
    
    for(int i = 0; i < targets.size(); i++) {
        if(targets[i][0] >= ref) {
            answer++;
            ref = targets[i][1];
        }
    }
    
    return answer;
}

File: programmers/LV2/요격 시스템.swift
import Foundation

// TODO: - 다시 풀어보기!

func solution(_ targets: [[Int]]) -> Int {
    let sortedTargets: [[Int]] = targets.sorted {
        if $0[1] == $1[1] {
            return $0[0] < $1[0]
        } else { 
            return $0[1] < $1[1]
        }
    }

    var answer: Int = 1
    var ref = sortedTargets[0][1]    
    
    for i in 1..<sortedTargets.count {
        if sortedTargets[i][0] >= ref {
            answer += 1
            ref = sortedTargets[i][1]
        }
    }
    
    return answer
}